### PR TITLE
Week 3: "/search" endpoint and model

### PR DESCRIPTION
**Pull request by enzofalone**
## Summary

- Created model feature SearchUsers which allows to query for different fields (budget, cleanliness, gender, coed, loudness)
- Created endpoint `/search` which executes function above requiring two query parameters, `query` and `field`
- Created file "user" in handler package for modularity

### CODE REVIEW

File: server/handlers/user.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+func SearchUsers(w http.ResponseWriter, res *http.Request) {
+	field := res.URL.Query().Get("field")
+	query := res.URL.Query().Get("query")
+
+	if len(field) == 0 || len(query) == 0 {
+		respondWithError(w, "Field/Query missing", http.StatusBadRequest)
+	}
+
+	users := data.SearchUsers(field, query)
```
Comment (lesterfernandez) [inline server/handlers/user.go:19]: Maybe we can rename this file to `search.go`?

File: server/data/user.go [hunk 2]
```diff
@@ -132,3 +141,40 @@ func IsValidLogin(username string, password string) bool {
 
 	return true
 }
+
+func SearchUsers(field string, query string) []*UserProfile {
```
Comment (lesterfernandez) [inline server/data/user.go:145]: This file is getting pretty big. Maybe we could move this function to a file called `data/search.go`, along with the `searchFields` map?

File: server/handlers/handlers.go [hunk 3]
```diff
@@ -32,7 +32,7 @@ func RegisterUser(w http.ResponseWriter, res *http.Request) {
 	}
 
 	err := data.CreateUser(newUser)
-	if err == nil {
+	if err != nil {
```
Comment (lesterfernandez) [inline server/handlers/handlers.go:35]: Nice job catching this bug! 
Now that I think about it, should these functions should be in `handlers/user.go`? You can just rename this file tbh.

**Review by lesterfernandez — CHANGES_REQUESTED** [review]
I'm very impressed with how fast you learned Go and were able to get a starting point for this endpoint. 

I left some suggestions for refactoring that I think you could squeeze into this PR as another commit.

File: server/handlers/handlers.go (hunk 3 above)
Comment (enzofalone) [reply server/handlers/handlers.go:35]: Sounds good! I think this way we will finally modularize the different components in the backend!

**Review by lesterfernandez — APPROVED** [review]

**Comment by lesterfernandez** [comment]
Nice dude!

**Pull request merged**